### DISCUSSION TITLE
RabbitMQ node stops right after installation

### DISCUSSION THREAD

**Discussion by MathewJoy2018** [rabbitmq/rabbitmq-server#12991] [Questions] [answered]
### Describe the bug

We install RMQ 3.10.6 and Erlang 25.0.3 as part  of our installation environment. On this Windows Server 2022 on Domain, the service wouldn't start. We upgraded (basically uninstall and reinstall) to 4.0.4 and 27.2 still get the same problem. I have the steps I followed below. In all our windows envs, it installed successfully. But on this customer server, it crashes as soon as it starts. I have the logs and the entire RMQ folder attached. 

### Reproduction steps

1. Uninstall RabbitMQ
2. Delete folder C:\Users\<user>\AppData\Roaming\RabbitMQ
3. Delete Program Files folders
4. Uninstall Erlang
5. Delete .erlang.cookie (typically found in %USERPROFILE%\.erlang.cookie), C:\Windows, C:\Windows\System32\config\systemprofile 
6. Delete Program Files folders
7. Reboot computer
8. Install Erlang 27 as Admin. May prompt for restarting computer.
9. Install RabbitMQ 4.0.4 as Admin
10. Ensure cookies are the same
11. Rabbitmqctl status

### Expected behavior

The RMQ service should start and not crash.

### Additional context

[RabbitMQ.zip](https://github.com/user-attachments/files/18213262/RabbitMQ.zip)

**Comment by lukebakken** [comment]
> But on this customer server, it crashes as soon as it starts

What is different about this server vs all others?

Please do the following:
* Log in as the user account you use to install RabbitMQ
* Open `cmd.exe`
* Run `set > set.txt` and _attach_ that file to this discussion.
* Change to the RabbitMQ server installation directory.
* Run `sbin\rabbitmq-server.bat`. Copy all of the output to a file, and _attach_ that file to this discussion.

**Comment by lukebakken** [comment]
Exception from log file:
```
error:function_clause

    filename:join/1, line 618
        args: [[]]
    filename:absname_vr/3, line 228
    rabbit:log_banner/0, line 1401
    rabbit:start/2, line 972
    application_master:start_it_old/4, line 295
```

**Comment by lukebakken** [comment, marked as answer]
The problem is that RabbitMQ can't determine the home directory for the user who installed RabbitMQ on this particular Windows machine.

This code:
https://github.com/rabbitmq/rabbitmq-server/blob/v4.0.x/deps/rabbit/src/rabbit.erl#L1401

...calls this code:
https://github.com/rabbitmq/rabbitmq-server/blob/v4.0.x/deps/rabbit/src/rabbit.erl#L1679-L1683

...which means that the Erlang VM can't determine the value for `home` when it starts. This is the code that determines the home directory on Windows:

https://github.com/erlang/otp/blob/maint-27/erts/etc/common/erlexec.c#L1629-L1651

On Windows, this code uses `HOMEDRIVE` and `HOMEPATH`, _or_ the value of `USERPROFILE`, to determine the home directory. This means that, on the system you're having trouble with, none of these environment variables are set.

The zip file you provided shows that there is a zero-byte file named `C` in the RabbitMQ directory. This is very suspicious, and supports the theory that there is something amiss with the user account you used to install RabbitMQ on the problematic system:

![image](https://github.com/user-attachments/assets/f8854dff-172d-404c-b5ee-c929be79d9c3)

**Comment by michaelklishin** [comment]
A mandatory mention of the fact that RabbitMQ 3.10.x has been [out of community support for more than two years](https://www.rabbitmq.com/release-information).
